experimental/PlaneCurve/ParaPlaneCurves.jl

```diff
@@ -0,0 +1,152 @@
+export parametrization_plane_curve, adjoint_ideal, rational_point_conic,
+       parametrization_conic, map_to_rational_normal_curve,
+       rat_normal_curve_anticanonical_map, rat_normal_curve_It_Proj_Odd,
+       rat_normal_curve_It_Proj_Even
+
+
+################################################################################
+function _tosingular(C::ProjPlaneCurve{fmpq})
+    F = C.eq
+    T = parent(F)
+    Tx = singular_ring(T)
+    return Tx(F)
+end
+
+function _fromsingular_ring(R::Singular.PolyRing)
+    Kx = base_ring(R)
+    if typeof(Kx) == Singular.N_AlgExtField
+        FF, t = RationalFunctionField(QQ, "t")
+        f = numerator(FF(Kx.minpoly))
+        K, _ = NumberField(f, "i")
+    else
+        K = QQ
+    end
+    newring, _ = PolynomialRing(K, [string(x) for x in gens(R)])
+    return newring
+end
+
+function _tosingular_ideal(C::ProjCurve)
+    I = C.I
+    singular_assure(I)
+    return I.gens.S
+end
+
+@doc Markdown.doc"""
+    parametrization_plane_curve(C::ProjPlaneCurve{fmpq}, s::String = "local")
+
+Return an ideal `P` in a new ring `R` consisting of a rational parametrization
+of the rational plane curve `C`. The generators of `P` parametrize the three
+coordinates of the rational curve. The ground field of `R` is either QQ or some
+algebraic extension QQ(a). The optional string can be "normal", to compute the
+integral basis via normalization, or local (default), to make local analysis of
+singularities first and apply normalization separately.
+"""
+function parametrization_plane_curve(C::ProjPlaneCurve{fmpq}, s::String = "local")
+    F = _tosingular(C)
+    L = Singular.LibParaplanecurves.paraPlaneCurve(F, s)
+    R = L[1]
+    J = [L[2][i] for i in keys(L[2])][1]
+    S = _fromsingular_ring(R)
+    return ideal(S, J)
+end
+
+@doc Markdown.doc"""
+    adjoint_ideal(C::ProjPlaneCurve{fmpq}, n::Int = 2, s::String= "")
+
+Return the adjoint ideal of the curve `C`. If n = 1, the computation is done via
+normalization, if n = 2 (default value), the local analysis of singularities is
+made first and then the normalization is applied separately, if n = 3 (resp
+n = 4), it uses normalization via (resp local) ideal quotient, and if n = 4.
+If the optional string "rattestyes" is used, it causes error message if the
+curve is not raional, and the string "firstchecksdone" prevents that check of
+assumptions will be done more than once.
```

Comment:
Please omit s::String= "" and the respective comment. This is not for the OSCAR User.